### ISSUE TITLE
Commit changes are highlighted just the opposite (in merge commit)

### ISSUE THREAD

**Issue by sijojlouis** [zaggino/brackets-git#949]
The commit history preview is showing wrong for me, for example:

` - <li><a href="#home">HOME</a>`
` + <li><a href="#home" class="navbuttons">HOME</a>`

should actually be (and the highlighted colors too)

` + <li><a href="#home">HOME</a>`
` - <li><a href="#home" class="navbuttons">HOME</a>`

--------------------------------

*Brackets 1.2.0-15697 (linux), Git 1.9.1 & Brackets-Git 0.14.17*

**Comment by zaggino** [comment]
Can you provide screenshots and debug logs for this please?

**Comment by sijojlouis** [comment]
Here is the screenshot of Brackets-Git
![brackets-git-shot](https://cloud.githubusercontent.com/assets/7536253/6540296/badbdae0-c4bc-11e4-908d-037179327d6a.jpg)

This is how the same code/commit is on Bitbucket ![bitbucket-shot](https://cloud.githubusercontent.com/assets/7536253/6540297/badbf4ee-c4bc-11e4-83c1-5fc355641590.jpg)

The Bitbucket highlights the changes correctly

**Comment by sijojlouis** [comment]
Here's the debug log:
```
Can't track online status, retry in 5 mins online.js:25
[brackets-nodejs] Connected to nodejs main.js:40
[brackets-git] Event invoked: brackets.project.change (1 listeners) Utils.js:337
NodeSocketTransport - start /LiveDevelopment/MultiBrowserImpl/transports/NodeSocketTransport.js:87
[brackets-git] cmd-spawn: /home/joelouis/.config/Brackets/extensions/user/zaggino.brackets-git/ -> git --version Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/.config/Brackets/extensions/user/zaggino.brackets-git/ -> /usr/local/git/bin/git --version Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/.config/Brackets/extensions/user/zaggino.brackets-git/ -> /usr/local/bin/git --version Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/.config/Brackets/extensions/user/zaggino.brackets-git/ -> /usr/bin/git --version Utils.js:342
[brackets-git] Event invoked: brackets.current.document.change - [object Object], [File /home/joelouis/git/bb/*****.com/index.html], first-pane (3 listeners) Utils.js:337
[brackets-git] cmd-spawn-fail (2578ms;+conn;ID=3): "ProcessUtils can't resolve the path requested: /usr/local/bin/git" Utils.js:342
[brackets-git] cmd-spawn-fail (2584ms;+conn;ID=2): "ProcessUtils can't resolve the path requested: /usr/local/git/bin/git" Utils.js:342
[brackets-git] cmd-spawn-out (2687ms;+conn;ID=4): "git version 1.9.1" Utils.js:342
[brackets-git] cmd-spawn-out (2694ms;+conn;ID=1): "git version 1.9.1" Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git status -u --porcelain Utils.js:342
[brackets-git] cmd-spawn-out (1213ms;ID=5) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git rev-list HEAD --not --remotes Utils.js:342
[brackets-git] Event invoked: git.status.results (3 listeners) Utils.js:337
[brackets-git] cmd-spawn-out (270ms;ID=6) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git rev-parse --show-toplevel Utils.js:342
[brackets-git] cmd-spawn-out (13ms;ID=7): "/home/joelouis/git/bb/*****.com" Utils.js:342
[brackets-git] Checking path for .git: /home/joelouis/git/bb/*****.com Utils.js:342
[brackets-git] Found .git in path: /home/joelouis/git/bb/*****.com Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git branch --no-color Utils.js:342
[brackets-git] cmd-spawn-out (34ms;ID=8): "* develop
  master" Utils.js:342
[brackets-git] Event invoked: rebase.merge.mode - false, false (1 listeners) Utils.js:337
[brackets-git] Event invoked: git.enabled (7 listeners) Utils.js:337
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git status -u --porcelain -- index.html Utils.js:342
[brackets-git] cmd-spawn-out (79ms;ID=9) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git config user.name Utils.js:342
[brackets-git] cmd-spawn-out (28ms;ID=10): "Sijo Louis" Utils.js:342
[brackets-git] Event invoked: git.username.changed - Sijo Louis (1 listeners) Utils.js:337
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git config user.email Utils.js:342
[brackets-git] cmd-spawn-out (33ms;ID=11): "s****@gmail.com" Utils.js:342
[brackets-git] Event invoked: git.email.changed - s****@gmail.com (1 listeners) Utils.js:337
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git remote -v Utils.js:342
[brackets-git] cmd-spawn-out (13ms;ID=12): "origin	https://*****@bitbucket.org/*****/*****.com.git (fetch)
origin	https://*****@bitbucket.org/*****/*****.com.git (push)" Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git status -u --porcelain Utils.js:342
[brackets-git] cmd-spawn-out (25ms;ID=13) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git rev-list HEAD --not --remotes Utils.js:342
[brackets-git] Event invoked: git.status.results (3 listeners) Utils.js:337
[brackets-git] cmd-spawn-out (28ms;ID=14) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git diff --no-ext-diff --no-color -U0 -- index.html Utils.js:342
[brackets-git] cmd-spawn-out (20ms;ID=15) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git config diff.tool Utils.js:342
[brackets-git] cmd-spawn-out (100ms;ID=16) Utils.js:342
[brackets-git] Event invoked: history.show - GLOBAL, [object Object] (1 listeners) Utils.js:337
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git branch --no-color Utils.js:342
[brackets-git] cmd-spawn-out (264ms;ID=17): "* develop
  master" Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git log -100 --format=%h_._%H_._%an_._%ai_._%ae_._%s_._%b_.nw._ develop -- Utils.js:342
[brackets-git] cmd-spawn-out (278ms;ID=18): "fbe91c0_._fbe91c06f84c6dc3d6d0188966f0f2341f6460a5_._Sijo J Louis_._2015-03-06 10:54:22 +0530_._s****@gmail.com_._Merge branch 'develop' of ssh://bitbucket.org/*****/*****.com into develop_.__.nw._
1951900_._195190018049a6a0e55bbbb9d41d929e2187c4fe_._Sijo Louis_._2015-03-06 10:51:58 +0530_._s****@gmail.com_._GA changes to record nav events_.__.nw._
26d3409_._26d340991e9f0a0845a85ad4c02ca8dfb831c81a_._Sijo Louis_._2015-03-06 10:17:20 +0530_._s****@gmail.com_._Content change_.__.nw._
f53b1b7_._f53b1b7d4060757c158060e72408e0812aad0b6c_._Sijo Louis_._2015-01-17 17:26:06 +0530_._s****@gmail.com_._added oracle_.__.nw._
9ff549a_._9ff549a50eab9a3a394d1df0f331795405338d56_._Sijo Louis_._2015-01-16 16:31:42 +0530_._s****@gmail.com_._Content changes to index_.__.nw._
2462a06_._2462a06dc1d7cd0a4a43b412a720c8f1309c24d3_._Sijo Louis_._2015-01-16 10:10:04 +0530_._s****@gmail.com_._gitignore file_.__.nw._
....
....
....
5a8b948_._5a8b9488ed04855efb0b88dcb59e3231329b5b86_._Sijo J Louis_._2014-06-18 00:46:53 +0530_._s****@gmail.com_._Initial commit_.__.nw._" Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git config diff.tool Utils.js:342
[brackets-git] cmd-spawn-out (42ms;ID=19) Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git diff --no-ext-diff --name-only fbe91c06f84c6dc3d6d0188966f0f2341f6460a5^! Utils.js:342
[brackets-git] cmd-spawn-out (65ms;ID=20): ".gitignore
index.html" Utils.js:342
[brackets-git] cmd-spawn: /home/joelouis/git/bb/*****.com/ -> git diff --no-ext-diff --no-color fbe91c06f84c6dc3d6d0188966f0f2341f6460a5^! -- index.html Utils.js:342
[brackets-git] cmd-spawn-out (76ms;ID=21): "diff --git a/index.html b/index.html
index 0cd9152..b551d9a 100644
--- a/index.html
+++ b/index.html
@@ -12,10 +12,10 @@
     <!-- Meta
     ================================================== -->
     <meta charset="utf-8">
-    <meta name="description" 
....
....
....

@@ -78,15 +78,15 @@
     <!--Main menu-->
     <nav class="mainnav">
         <ul>
-            <li><a href="#home">HOME</a>
+            <li><a href="#home" class="navbuttons">HOME</a>
             </li>
-            <li><a href="#about">ABOUT</a>
+            <li><a href="#about" class="navbuttons">ABOUT</a>
             </li>
-            <li><a href="#portfolio">PORTFOLIO</a>
+            <li><a href="#portfolio" class="navbuttons">PORTFOLIO</a>
             </li>
-            <li><a href="#contact">CONTACT</a>
+            <li><a href="#contact" class="navbuttons">CONTACT</a>
             </li>
-            <!-- <li><a href="blog.html">BLOG</a>
+            <!-- <li><a href="blog.html" class="navbuttons">BLOG</a>
             </li> -->
         </ul>
 
@@ -116,7 +116,7 @@
....
....
....
             </footer>
             <!--container-->" Utils.js:342
Can't track online status, retry in 5 mins online.js:25

```

**Comment by zaggino** [comment]
Strange, does this happen only on merge commits or on all of them?

**Comment by zaggino** [comment]
Diff results are displayed correctly, it seems we're calling the diff command in a wrong way for this certain commit... not sure what's wrong...

**Comment by sijojlouis** [comment]
Good question - it shows up correctly for the individual commit, but not for merge commit... so the issue seems to be with merge commit

**Comment by MarcelGerber** [comment]
@zaggino I've found such a merge commit to reproduce, too:
On GitHub:
https://github.com/adobe/brackets/commit/555e7b9b9f632d65ffad1322d4d7422b588228d8

In Brackets:
![image](https://cloud.githubusercontent.com/assets/2641501/6811181/6568467c-d266-11e4-8e12-bb35c66c46b4.png)

Investigating.

**Comment by zaggino** [comment]
This should be fixed in latest version `0.14.19`, please test/confirm. Thanks!
